src/codemagic/tools/app_store_connect/mixins/signing_file_saver_mixin.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import annotations
+
+import pathlib
+import re
+import tempfile
+from typing import List
+from typing import Optional
+from typing import Sequence
+
+from codemagic.apple.resources import Profile
+from codemagic.apple.resources import SigningCertificate
+from codemagic.models import Certificate
+from codemagic.models import PrivateKey
+
+from ..errors import AppStoreConnectError
+from ..resource_printer import ResourcePrinter
+
+
+class SigningFileSaverMixin:
+    certificates_directory: pathlib.Path
+    printer: ResourcePrinter
+    profiles_directory: pathlib.Path
+
+    @classmethod
+    def _get_unique_path(cls, file_name: str, destination: pathlib.Path) -> pathlib.Path:
+        if destination.exists() and not destination.is_dir():
+            raise ValueError(f"Destination {destination} is not a directory")
+        destination.mkdir(parents=True, exist_ok=True)
+        name = pathlib.Path(re.sub(r"[^\w.]", "_", file_name))
+        tf = tempfile.NamedTemporaryFile(
+            prefix=f"{name.stem}_",
+            suffix=name.suffix,
+            dir=destination,
+            delete=False,
+        )
+        tf.close()
+        return pathlib.Path(tf.name)
+
+    def _save_profile(self, profile: Profile) -> pathlib.Path:
+        profile_path = self._get_unique_path(
+            f"{profile.attributes.profileType}_{profile.id}{profile.profile_extension}",
+            self.profiles_directory,
+        )
+        profile_path.write_bytes(profile.profile_content)
+        self.printer.log_saved(profile, profile_path)
+        return profile_path
+
+    def _save_certificate(
```

Comment:
Originally defined in https://github.com/codemagic-ci-cd/cli-tools/blob/3670fa624997f17cb437bc88198f7bdee3594f35/src/codemagic/tools/app_store_connect.py#L1053-L1076